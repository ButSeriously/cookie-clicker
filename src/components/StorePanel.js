import React from 'react';
import PropTypes from 'prop-types';
import Building from './Building';
import BuildingInfo from './BuildingInfo';
import buildings from '../settings/buildings';

const StorePanel = props => (
  <div className="store_panel">
    <h1 className="store_panel__header">Store</h1>
    <div className="store_panel__buildings">
      <div className="store_panel__separator">
        <span>Buildings</span>
      </div>
      { buildings.map(building => (<Building
        key={building.name}
        data={building}
        cookiesAmount={props.cookiesAmount}
        buildingBought={props.buildingBought}
      />)) }
    </div>
    <BuildingInfo />
  </div>
);

StorePanel.propTypes = {
  cookiesAmount: PropTypes.number.isRequired,
  buildingBought: PropTypes.func.isRequired,
};

export default StorePanel;
