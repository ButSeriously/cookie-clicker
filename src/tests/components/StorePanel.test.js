import React from 'react';
import { shallow } from 'enzyme';
import StorePanel from '../../components/StorePanel';

test('should render StorePanel', () => {
  const wrapper = shallow(<StorePanel />);
  expect(wrapper).toMatchSnapshot();
});
