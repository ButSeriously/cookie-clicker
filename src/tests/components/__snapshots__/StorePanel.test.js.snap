// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render StorePanel 1`] = `
<div
  className="store_panel"
>
  <h1
    className="store_panel__header"
  >
    Store
  </h1>
  <div
    className="store_panel__buildings"
    onMouseLeave={[Function]}
  >
    <div
      className="store_panel__separator"
    >
      <span>
        Buildings
      </span>
    </div>
    <Building
      buildingBought={[Function]}
      cookiesAmount={100}
      data={
        Object {
          "description": "Autoclicks once every 10 seconds.",
          "initialCost": 15,
          "name": "Cursor",
          "productionPerSecond": 0.1,
        }
      }
      getInfoPanel={[Function]}
      key="Cursor"
    />
    <Building
      buildingBought={[Function]}
      cookiesAmount={100}
      data={
        Object {
          "description": "A nice grandma to bake more cookies.",
          "initialCost": 100,
          "name": "Grandma",
          "productionPerSecond": 1,
        }
      }
      getInfoPanel={[Function]}
      key="Grandma"
    />
    <Building
      buildingBought={[Function]}
      cookiesAmount={100}
      data={
        Object {
          "description": "Grows cookie plants from cookie seeds.",
          "initialCost": 1100,
          "name": "Farm",
          "productionPerSecond": 8,
        }
      }
      getInfoPanel={[Function]}
      key="Farm"
    />
    <Building
      buildingBought={[Function]}
      cookiesAmount={100}
      data={
        Object {
          "description": "Mines out cookie dough and chocolate chips.",
          "initialCost": 12000,
          "name": "Mine",
          "productionPerSecond": 47,
        }
      }
      getInfoPanel={[Function]}
      key="Mine"
    />
    <Building
      buildingBought={[Function]}
      cookiesAmount={100}
      data={
        Object {
          "description": "Produces large quantities of cookies.",
          "initialCost": 130000,
          "name": "Factory",
          "productionPerSecond": 260,
        }
      }
      getInfoPanel={[Function]}
      key="Factory"
    />
  </div>
  <BuildingInfo
    buildingInfo={Object {}}
  />
</div>
`;
